### ISSUE TITLE
[0.9.23] BPopOver "Hide/Show" unclear

### ISSUE THREAD

**Issue by susanne99** [bootstrap-vue-next/bootstrap-vue-next#1275]
i read the docu:
https://bootstrap-vue-next.github.io/bootstrap-vue-next/docs/components/popover.html

there is a prop "show" but it is not public (browser vue dev tools doesn't show it). instead there is a public  a prop "hide".
according to the docu from bootstrap-vue, there should be both props "hide" & "show" avaiable.

besides that, the exisiting prop "hide" does nothing. 
i would like close the popup when the user clicks on the "popup component".
background: i have text with tooltip made with popup, but when the user click on the text, a modal dialog opens and the tooltip keeps open. therefore, when i open the dialog i will close the popover tooltip .

or, another example:
the popover keeps open, when i drag&drop THE popover element, for instance a calendar appointment. during drg&drop the mouse is pushed and no "hover out" occurs and that's why it stays open. i know when drag&drop starts and could close the popup, but the "hide" prop doesn't working

how can i make this requirement

**Comment by xvaara** [comment]
The documentation is a bit unclear. The hide prop is for a middleware for hiding the popover when the reference leaves viewport: https://floating-ui.com/docs/hide

show prop is to show the popover when it's created. They are not reactive. If you need to control the popover you can just pass a v-model to it and change that to false to hide the popover.

**Comment by xvaara** [comment]
> +1 Attempted to use `v-model` to programmatically close the popover using a `b-button` which didn't seem to work

Show me the code. Here's a working example: https://stackblitz.com/edit/vue-n7cn3j?file=src%2FApp.vue,src%2Fcomponents%2FComp.vue

**Comment by mediaessenz** [comment]
The example with the v-model is working for a single popopver, but how can this be done, if I have multible popovers generated by a for loop?

If I use a single v-model for all of them, all popovers appears/disappears if I change it's state.

I need a way to close programatically a specific popover.

In the past of bootstrap-vue, I was able to close an open popovers with this emit:
`this.$root.$emit('bv::hide::popover', id);` 

How can this be done in bootstrap-vue-next?

**Comment by VividLemon** [comment]
@xvaara Hypothetically you could modify the code for useModal and adapt it to behave for tooltip and popover. Tbh I would just combine the two into one function. I don't see a reason why they would be separated. Tooltips and popovers are too similar to warrant different functions.

**Comment by xvaara** [comment]
> The example with the v-model is working for a single popopver, but how can this be done, if I have multible popovers generated by a for loop?
> 
> If I use a single v-model for all of them, all popovers appears/disappears if I change it's state.
> 
> I need a way to close programatically a specific popover.
> 
> In the past of bootstrap-vue, I was able to close an open popovers with this emit: `this.$root.$emit('bv::hide::popover', id);`
> 
> How can this be done in bootstrap-vue-next?

you can use an object? or use template refs? 
https://stackblitz.com/edit/vue-sjmqrm?file=src%2Fcomponents%2FComp.vue

**Comment by xvaara** [comment]
> @xvaara Hypothetically you could modify the code for useModal and adapt it to behave for tooltip and popover. Tbh I would just combine the two into one function. I don't see a reason why they would be separated. Tooltips and popovers are too similar to warrant different functions.

@VividLemon I see, this could be generalised as a toggleable composable, also used for collapse, dropdown (are there any other components that have toggle/show/close? Or is this needed?

**Comment by VividLemon** [comment]
> > @xvaara Hypothetically you could modify the code for useModal and adapt it to behave for tooltip and popover. Tbh I would just combine the two into one function. I don't see a reason why they would be separated. Tooltips and popovers are too similar to warrant different functions.
> 
> @VividLemon I see, this could be generalised as a toggleable composable, also used for collapse, dropdown (are there any other components that have toggle/show/close? Or is this needed?

If we were to do this, we should have it written down about which things are able to be toggled with exposed methods, which are able to be toggled with directives, and which are able to be toggled with composables (in addition to the vmodel, but that goes without saying). Having each of these things hidden around is getting confusing.

**Comment by KenEucker** [comment]
I was able to programmatically close a b-popover by calling .hide with a parameter of { type: 'click' }, which does, in fact, close the popover. [See Reference.](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/d1121da3ea75aea20e16e1892070fba89a72949a/packages/bootstrap-vue-next/src/components/BPopover.vue#L394)

```
popoverRef.hide({ type: 'click' })
```

This does not work, but should it?
```
popoverRef.hide()
```

**Comment by hackel** [comment]
I also had trouble using the hide method, but the toggle method worked.  So I copied the code from the toggle method, which is `hide(new Event('click'))`.  It would be nice if the hide method did this for you if you call it with no arguments.

**Comment by xvaara** [comment]
Yes, we should check if the hide is called with no arguments and just hide. Oversight on my part. Should not cause any side effects.

**Comment by RmK9** [comment]
> Yes, we should check if the hide is called with no arguments and just hide. Oversight on my part. Should not cause any side effects.

Hey @xvaara was this updated recently? Specifying triggers/type seems to be throwing an error now, however, it works without the need to specify anymore which is even better!
